src/test/java/org/sagebionetworks/bridge/services/AdherenceServiceTest.java

```diff
@@ -1383,6 +1396,340 @@ public void getAdherenceStatistics_adherenceTooHigh() {
         service.getAdherenceStatistics(TEST_APP_ID, TEST_STUDY_ID, 200);
     }
     
+    @Test
+    public void getDetailedAdherenceReportForParticipant() {
+        AdherenceRecord assessmentRecord1 = ar(STARTED_ON, FINISHED_ON, "assessment-instance-guid-1", false);
+        assessmentRecord1.setUploadedOn(UPLOADED_ON);
+        AdherenceRecord sessionRecord1 = sar(STARTED_ON, FINISHED_ON, "session-instance-guid-1", false);
+
+        AdherenceRecord assessmentRecord2 = ar(STARTED_ON, FINISHED_ON, "assessment-instance-guid-2", false);
+        AdherenceRecord studyBurstRecord1 = sar(STARTED_ON, FINISHED_ON, "study-burst-session-guid-2", false);
+        
+        Account account = mockDetailedAdherenceTimeline(ImmutableList.of(assessmentRecord1, sessionRecord1, 
+                assessmentRecord2, studyBurstRecord1));
+        
+        
+        DetailedAdherenceReport report = service.getDetailedAdherenceReportForParticipant(
+                TEST_APP_ID, TEST_STUDY_ID, account);
+        
+        assertEquals(report.getParticipant().getIdentifier(), TEST_USER_ID);
+        assertEquals(report.getParticipant().getExternalId(), TEST_EXTERNAL_ID);
+        assertTrue(report.isTestAccount());
+        assertEquals(report.getJoinedDate().withZone(DateTimeZone.forID("America/Los_Angeles")).toString(),
+                ENROLLMENT.toString());
+        
+        List<DetailedAdherenceReportSessionRecord> sessionRecords = report.getSessionRecords();
+        assertEquals(sessionRecords.size(), 2);
```

Comment:
Should we also check the sort priority of the two session records and their assessment records?

Reply:
I added another assessment to get an assessment order check in. The sessions and bursts are both just ScheduledSessions in the timeline so their order is getting checked as well.